h2o-algos/src/main/java/hex/pca/PCA.java

```diff
@@ -58,26 +56,28 @@
  //   int numCPUs= H2O.NUMCPUS;   // proper way to get number of CPUs.
     double p = hex.util.LinearAlgebraUtils.numColsExp(_train,true);
     double r = _train.numRows();
-    long mem_usage =
-            _parms._pca_method == PCAParameters.Method.GramSVD ? (long)(hb._cpus_allowed * p*p * 8/*doubles*/ *
+    boolean useGramSVD = _parms._pca_method == PCAParameters.Method.GramSVD;
+    boolean usePower = _parms._pca_method==PCAParameters.Method.Power;
+
+    long mem_usage = (useGramSVD || usePower) ? (long)(hb._cpus_allowed * p*p * 8/*doubles*/ *
                     Math.log((double)_train.lastVec().nChunks())/Math.log(2.)) : 1; //one gram per core
-    long mem_usage_w = _parms._pca_method == PCAParameters.Method.GramSVD ? (long)(hb._cpus_allowed * r*r *
+    long mem_usage_w = (useGramSVD || usePower) ? (long)(hb._cpus_allowed * r*r *
```

Comment:
In GramSVD, you form the gram matrix and proceed to take its SVD to derive your eigenvalues, eigenvectors.

In power method, you form your gram matrix, however you do not take its SVD to derive the eigenvalues, eigenvectors.  Instead, you keep multiplying your gram matrix with a random initial vector x0 until you hit a stopping condition and viola, you got your eigenvectors.  Essentially, you are taking the nth power of your gram matrix and hence the name of the algo.

 Here is a link to a good illustration: http://college.cengage.com/mathematics/larson/elementary_linear/5e/students/ch08-10/chap_10_3.pdf.